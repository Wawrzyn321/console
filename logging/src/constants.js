export const SORT_DESCENDING = 'descending';
export const SORT_ASCENDING = 'ascending';

export const SORT_TYPES = [SORT_DESCENDING, SORT_ASCENDING];

export const PERIODS = [
  'last minute',
  'last 5 minutes',
  'last 15 minutes',
  'last hour',
];

export const DEFAULT_PERIOD = PERIODS[2];
