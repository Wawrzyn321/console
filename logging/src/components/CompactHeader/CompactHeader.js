import React from 'react';
import PropTypes from 'prop-types';
import './CompactHeader.scss';

import { Panel } from 'fundamental-react';

import SearchInput from './../Shared/SearchInput/SearchInput';
import OptionsDropdown from '../Shared/SelectDropdown/SelectDropdown';
import { PERIODS, SORT_TYPES } from '../../constants';
import ResultOptionsDropdown from '../Shared/ResultsOptionsDropdown/ResultsOptionsDropdown';

CompactHeader.propTypes = {
  updateFilteringState: PropTypes.func.isRequired,
  searchPhrase: PropTypes.string.isRequired,
  logsPeriod: PropTypes.oneOf(PERIODS),
  sortDirection: PropTypes.oneOf(SORT_TYPES),
  advancedSettings: PropTypes.object.isRequired,
};

export default function CompactHeader({
  updateFilteringState,
  searchPhrase,
  logsPeriod,
  sortDirection,
  advancedSettings,
}) {
  function updateLogsPeriod(logsPeriod) {
    updateFilteringState({ logsPeriod });
  }

  function updateSortDirection(sortDirection) {
    updateFilteringState({ sortDirection });
  }

  return (
    <Panel className="fd-has-padding-regular fd-has-padding-bottom-tiny">
      <section className="compact-header">
        <h1 className="fd-has-type-3">Logs</h1>
        <div className="fd-has-display-flex">
          <SearchInput
            searchPhrase={searchPhrase}
            updateFilteringState={updateFilteringState}
            compact={true}
          />
          <ResultOptionsDropdown
            advancedSettings={advancedSettings}
            updateFilteringState={updateFilteringState}
          />
          <OptionsDropdown
            availabelValues={PERIODS}
            currentValue={logsPeriod}
            icon="past"
            updateValue={updateLogsPeriod}
            compact={true}
          />
          <OptionsDropdown
            availabelValues={SORT_TYPES}
            currentValue={sortDirection}
            icon="sort"
            updateValue={updateSortDirection}
            compact={true}
          />
        </div>
      </section>
    </Panel>
  );
}
