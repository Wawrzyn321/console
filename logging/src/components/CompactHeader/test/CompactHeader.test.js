import React from 'react';
import renderer from 'react-test-renderer';
import CompactHeader from './../CompactHeader';
import { DEFAULT_PERIOD, SORT_ASCENDING } from '../../../constants';

describe('CompactHeader', () => {
  console.error = jest.fn();

  afterEach(() => {
    console.error.mockReset();
  });

  afterAll(() => {
    expect(console.error.mock.calls[0][0]).toMatchSnapshot(); // unique "key" prop warning
  });

  it('Renders with minimal props', () => {
    const component = renderer.create(
      <CompactHeader
        updateFilteringState={() => {}}
        searchPhrase="search phrase"
        logsPeriod={DEFAULT_PERIOD}
        sortDirection={SORT_ASCENDING}
        advancedSettings={{}}
        autoRefreshEnabled={false}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
