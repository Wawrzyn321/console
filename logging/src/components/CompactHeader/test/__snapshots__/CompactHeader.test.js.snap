// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactHeader Renders with minimal props 1`] = `
<div
  className="fd-panel fd-has-padding-regular fd-has-padding-bottom-tiny"
>
  <section
    className="compact-header"
  >
    <h1
      className="fd-has-type-3"
    >
      Logs
    </h1>
    <div
      className="fd-has-display-flex"
    >
      <section
        className="fd-has-margin-right-small"
      >
        <span
          className="caption-muted search-input__caption"
        />
        <div
          className="fd-search-input search-input--compact"
          value="search phrase"
        >
          <div
            className="fd-popover"
          >
            <div
              className="fd-popover__control"
            >
              <div
                aria-expanded={false}
                aria-haspopup="true"
                className="fd-combobox-control"
              >
                <div
                  className="fd-input-group fd-input-group--after"
                >
                  <input
                    className="fd-input"
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Search"
                    style={Object {}}
                    type="text"
                    value=""
                  />
                  <span
                    className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                  >
                    <button
                      className="fd-button--light sap-icon--search"
                      onClick={[Function]}
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <span
        className="link-button fd-has-type-minus-1"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--action-settings fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              />
            </div>
          </div>
        </div>
      </span>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--past fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              />
            </div>
          </div>
        </div>
      </span>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--sort fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              />
            </div>
          </div>
        </div>
      </span>
    </div>
  </section>
</div>
`;
